scrapy/spiders/__init__.py

```diff
@@ -17,12 +17,17 @@
 from scrapy.utils.url import url_is_from_spider
 
 if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    # typing.Concatenate requires Python 3.10
     # typing.Self requires Python 3.11
-    from typing_extensions import Self
+    from typing_extensions import Concatenate, Self
 
     from scrapy.crawler import Crawler
     from scrapy.settings import BaseSettings
 
+    CallbackT = Callable[Concatenate[Response, ...], Any]
```

Comment:
This could be useful in other places as it's a general type for a spider callback. I didn't think about that further though.

Reply:
Actually we should use it for `Request.callback`

Reply:
Looks like we can't, probably because of wrong covariance when assigning a specific callback to a request, I need to investigate this separately.